### DISCUSSION TITLE
Who can share me what is best practice when using nest.js + prisma + graphql federation

### DISCUSSION THREAD

**Discussion by worapolburaphan** [prisma/prisma#11238] [Q&A] [answered]
How to fit grpahql schema with prisma model

I spend a most of time in my work with this problem but I just nearly the answer but it isn't work. I want to query graphql like query prisma on my client and shield it with authentication

I try to using typegraphql it's not working on federation 1 & 2
I try to using nexus but it can't using with nest.js
I try to using code first way it not work because type is not compat with prisma anymore

what can I do with my use case

**Comment by pantharshit00** [comment, marked as answer]
Hello!

Sorry for very late response here. We are currently reviewing past discussions and saw this was not answered. 

There is an example in the official typegraphql repository which can help you here: https://github.com/MichalLytek/type-graphql/tree/master/examples/apollo-federation

You can also ask in the respective libraries for support here because federation is something that happens on the graphql library level. Only things that we can do here is bug fixes for types so please report anything you find in that regard.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
